### PR TITLE
fix(api): forbid using `asc`/`desc` in selections

### LINKED ISSUE #8080: `table.mutate(c = _.a.desc())` differs across implementations
### What happened?

```python
In [2]: import ibis
   ...: from ibis import _
   ...:
   ...: con = ibis.pandas.connect()
   ...: con.create_table("table", ibis.memtable({"a": [1, 2, 3], "b": [4, 5, 6]}))
   ...: table = con.table("table")
   ...:
   ...: table.mutate(c = _.a.desc()).execute()
Out[2]:
   a  b  c
0  1  4  1
1  2  5  2
2  3  6  3

In [3]: import ibis
   ...: from ibis import _
   ...:
   ...: con = ibis.polars.connect()
   ...: con.create_table("table", ibis.memtable({"a": [1, 2, 3], "b": [4, 5, 6]}))
   ...: table = con.table("table")
   ...:
   ...: table.mutate(c = _.a.desc()).execute()
Out[3]:
   a  b  c
0  1  4  3
1  2  5  2
2  3  6  1
```

Looks like the results differ? (if I could hazard a guess - is pandas' index autoaligning something under the hood?)

### What version of ibis are you using?

7.2.0

### What backend(s) are you using, if any?

pandas / polars

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by jcrist**
Fixes #8080.

### CODE REVIEW

File: ibis/expr/operations/relations.py [hunk 1]
```diff
@@ -171,7 +172,7 @@ def _check_integrity(values, allowed_parents):
 @public
 class Project(Relation):
     parent: Relation
-    values: FrozenDict[str, Unaliased[Value]]
+    values: FrozenDict[str, NonSortKey[Unaliased[Value]]]
```
Comment (jcrist) [inline ibis/expr/operations/relations.py:175]: There may be other places we want to add this restriction, but I didn't want to bother with them in this PR for now.

**Comment by kszucs** [comment]
I have been thinking about introducing a `Marker` base class for wrapping value operations with additional metadata, like `alias` or `ascending`, then we could use an `UnmarkedValue` uniformly across the annotations. Going to think about it.

**Review by kszucs — APPROVED** [review]

**Comment by jcrist** [comment]
Since aliases are valid in many places but `asc()`/`desc()` are really only valid in `order_by()` I wonder if `asc()`/`desc()` should return non `Value` operations? This would make it easier to forbid e.g. `t.x.asc() + 1`.

**Comment by kszucs** [comment]
Yes, that should be the correct approach.

**Pull request merged**